components/Modal.tsx

```diff
@@ -0,0 +1,80 @@
+import { Dialog, Transition } from "@headlessui/react";
+import { forwardRef, Fragment, useImperativeHandle, useState } from "react";
+import { XMarkIcon } from "@heroicons/react/20/solid";
+
+interface Modal {
+  open: () => void;
+  close: () => void;
+}
+
+interface Props {
+  title?: string;
+  children: React.ReactNode;
+}
+
+const Modal = forwardRef<Modal, Props>(({ title, children }, ref) => {
```

Comment:
Why do we need to use a ref and forward ref here?

Reply:
![Screenshot 2023-02-24 at 17 59 44 (2)](https://user-images.githubusercontent.com/68110533/221254932-5bb6ebe0-5f3a-40ee-ae54-9309f9c86652.png)

Reply:
It looks like you are only using the ref for the modal state, you should just be able to use state props.

Reply:
honestly this is a copy paste from some code ben wrote lol. I dont really know why it's using ref. 
I'll refactor it.